src/main/java/net/glease/structurecompat/CompatBloodMagic.java

```diff
@@ -0,0 +1,158 @@
+                    {"i                     i",T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,"i                     i"},
+                    {"i                     i",T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,T6_S,"i                     i"},
+                    {"  rrrrrrrrrrrrrrrrrrr  ",T6_S,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_SR,T6_S, "  rrrrrrrrrrrrrrrrrrr  "},
+                }))
+            .addElement('a',ofBlock(ModBlocks.blockAltar,0))
+            .addElement('r',ofChain(ofBlockAnyMeta(ModBlocks.bloodRune),ofBlockAnyMeta(ModBlocks.speedRune),ofBlockAnyMeta(ModBlocks.runeOfSelfSacrifice),ofBlockAnyMeta(ModBlocks.runeOfSacrifice),ofBlockAnyMeta(ModBlocks.efficiencyRune)))
+            .addElement('g',ofBlock(AlchemicalWizardry.specialAltarBlock[1].getBlock(),0))
+            .addElement('t',ofBlock(AlchemicalWizardry.specialAltarBlock[0].getBlock(),0))
+            .addElement('b',ofBlock(AlchemicalWizardry.specialAltarBlock[3].getBlock(),0))
+            .addElement('v',ofBlock(AlchemicalWizardry.specialAltarBlock[2].getBlock(),0))
+            .addElement('e',ofBlock(AlchemicalWizardry.specialAltarBlock[4].getBlock(),0))
+            .addElement('c',ofBlock(AlchemicalWizardry.specialAltarBlock[6].getBlock(),0))
+            .addElement('i',ofBlock(AlchemicalWizardry.specialAltarBlock[5].getBlock(),0))
+            .build();
+        IMultiblockInfoContainer.registerTileClass(
+            TEAltar.class,
+            new CompatBloodMagic.AltarMultiblockInfoContainer(structureAltar));
+        // spotless:on
+    }
+
+    private static ExtendedFacing noSideWay(ExtendedFacing aSide) {
+        return aSide.getDirection().offsetY != 0 ? ExtendedFacing.DEFAULT : aSide;
+    }
+
+    private static class AltarMultiblockInfoContainer implements IMultiblockInfoContainer<TEAltar> {
+
+        private final IStructureDefinition<TEAltar> structureAltar;
+
+        public AltarMultiblockInfoContainer(IStructureDefinition<TEAltar> structureAltar) {
+            this.structureAltar = structureAltar;
+        }
+
+        @Override
+        public void construct(ItemStack triggerStack, boolean hintsOnly, TEAltar ctx, ExtendedFacing aSide) {
+            int tier = triggerStack.stackSize;
+            if (tier > 6) {
+                tier = 6;
+            }
+            for (int i = 1; i <= tier; i++) {
+                System.out.println("tier" + i);
+                this.structureAltar.buildOrHints(
+                        ctx,
+                        triggerStack,
+                        "tier" + i,
+                        ctx.getWorldObj(),
+                        noSideWay(aSide),
```

Comment:
There is only one orientation of this multi. Just use ExtendedFacing.DEFAULT regardless of what was given.